lib/pf/detect/parser/suricata_md5.pm

```diff
@@ -28,38 +28,60 @@ sub parse {
 
-    # Received line should be JSON encoded
-    $line =~ s/^.*?{/{/s;
-    my $data = decode_json($line);
+    my $data = $self->_parse($line);
 
     # Extracting basic information required to process any further
     # MD5 hash is mainly what we are working with
     if ( !defined($data->{'md5'}) || !$data->{'md5'} ) {
-        $logger->debug("Trying to parse a Suricata HTTP line that does not contain a MD5 hash value. Nothing to do");
+        $logger->debug("Trying to parse a Suricata md5 file line that does not contain a MD5 hash value. Nothing to do");
         return 0;   # Returning 0 to pfdetect indicates "job's done"
     }
-    my $md5_hash = $data->{'md5'};
+    
+    my $endpoint;
+    #We want to know what is the protocol that was linked to the md5 file extraction. 
+    #For that protocol, determine if the endpoint is dstip or srcip
+    #For http (http_host), etc, possible infected endpoint is the dstip
+    if ( defined $data->{'http_host'} && $data->{'http_host'} ) {
+        $endpoint='dstip';
+    }
+    
+    #For smtp (sender), etc, possible infected endpoint is the srcip
+    elsif ( defined $data->{'sender'} && $data->{'sender'} ) { 
+        $endpoint='srcip';
+    }
 
-    # Destination IP is the client IP address on which we want to act and from which we get the MAC address
-    if ( !defined($data->{'dstip'}) || !$data->{'dstip'} ) {
-        $logger->debug("Trying to parse a Suricata HTTP line that does not contain a destination IP. Nothing to do");
+    # endpoint IP is the client IP address on which we want to act and from which we get the MAC address
+    if ( !defined($data->{$endpoint}) || !$data->{$endpoint} ) {
+        $logger->debug("Trying to parse a Suricata md5 file line that does not contain a destination IP. Nothing to do");
         return 0;   # Returning 0 to pfdetect indicates "job's done"
     }
-    my $mac = pf::iplog::ip2mac($data->{'dstip'});
+
+    my $mac = pf::iplog::ip2mac($data->{$endpoint});
     if ( !defined($mac) || !$mac ) {
-        $logger->debug("Trying to parse a Suricata HTTP line without a valid client MAC address. Nothing to do");
+        $logger->debug("Trying to parse a Suricata md5 file line without a valid client MAC address. Nothing to do");
         return 0;   # Returning 0 to pfdetect indicates "job's done"
     }
     $data->{'mac'} = $mac;  # Adding processed MAC address to the data hash to avoid having to processing it again later on
 
     # We do the process async using API rather than going through pfdetect since it requires external resources evaluation that may take some time
     my $apiclient = pf::api::queue->new;
-    $apiclient->notify('trigger_violation', ( 'mac' => $mac, 'tid' => $md5_hash, 'type' => 'suricata_md5' ));   # Process Suricata MD5 based violations
+    $apiclient->notify('trigger_violation', ( 'mac' => $data->{mac}, 'tid' => $data->{md5}, 'type' => 'suricata_md5' ));   # Process Suricata MD5 based violations
```

Comment:
He adds it in the hash (line 64) using the result of ip2mac lookup, this isn't from the parsed line, so I guess we could say it will always be there :)